rct229/rules/section22/section22rule19.py

```diff
@@ -0,0 +1,74 @@
+from rct229.data.schema_enums import schema_enums
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.utils.jsonpath_utils import find_all, find_exactly_one_with_field_value
+
+APPLICABLE_SYS_TYPES = [
+    "SYS-7",
+    "SYS-8",
+    "SYS-11.1",
+    "SYS-11.2",
+    "SYS-12",
+    "SYS-13",
+    "SYS-7B",
+    "SYS-8B",
+    "SYS-11B",
+    "SYS-12B",
+    "SYS-13B",
+]
+REQUIRED_TEMP_RESET = schema_enums["TemperatureResetOptions"]
+
+
+class Section22Rule19(RuleDefinitionListIndexedBase):
+    """Rule 19 of ASHRAE 90.1-2019 Appendix G Section 22 (Chilled water loop)"""
+
+    def __init__(self):
+        super(Section22Rule19, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section22Rule19.HeatRejectionRule(),
+            index_rmr="baseline",
+            id="22-19",
+            description="The tower shall be controlled to maintain a leaving water temperature, where weather permits.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="heat_rejections[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        # FIXME: replace with baseline_system_types = get_baseline_system_types(rmi_b) when get_baseline_system_types
+        #  is ready.
+        baseline_system_types = {
+            "SYS-7": ["hvac_sys_7"],
+            "SYS-11B": ["hvac_sys_11_b"],
+        }
+        # if any system type found in the APPLICABLE_SYS_TYPES then return applicable.
+        return any(
+            [key in APPLICABLE_SYS_TYPES for key in baseline_system_types.keys()]
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        temp_reset_type_dict_b = {
+            heat_rejection_loop: find_exactly_one_with_field_value(
+                "$..fluid_loops[*]", "id", heat_rejection_loop, rmi_b
+            )["cooling_or_condensing_design_and_control"]["temperature_reset_type"]
```

Comment:
We should discuss how best to handle cases where the `cooling_or_condensing_desing_and_control` or `temperature_rest_type` fields are missing. I believe these should be required fields. I believe that `loop` should also be a required field for all `heat_rections[*]`.